src/web/client/extension.ts

```diff
@@ -268,12 +269,95 @@ export function processWillSaveDocument(context: vscode.ExtensionContext) {
     );
 }
 
-export function processWillStartCollaboartion() {
+export function processWillStartCollaboartion(context: vscode.ExtensionContext) {
     if (isCoPresenceEnabled()) {
-        // TODO: Add copresence logic
+        processOnTabChange(context);
+        createWebWorkerInstance(context);
     }
 }
 
+let worker: Worker;
+
+export function processOnTabChange(context: vscode.ExtensionContext) {
+    try {
+        context.subscriptions.push(
+            vscode.window.tabGroups.onDidChangeTabs((event) => {
+                event.opened.concat(event.changed).forEach(tab => {
+                    if (tab.input instanceof vscode.TabInputCustom || tab.input instanceof vscode.TabInputText) {
+                        const document = tab.input;
+                        const entityId = getFileEntityId(document.uri.fsPath);
+                        if (entityId) {
+                            worker.postMessage({
```

Comment:
worker.postmessage logic can go insideprocessWorkspaceStateChanges() method. Put this block inside if condition "false && isCoPresenceEnabled()". Its not a good idea to register same event twice.

Reply:
Done: https://github.com/microsoft/powerplatform-vscode/pull/745/commits/6df6477b87bbc533ffa00367a124f8a8fecda90a